applications/main/nfc/views/dict_attack.c

```diff
@@ -49,28 +41,29 @@ static void dict_attack_draw_callback(Canvas* canvas, void* model) {
                 "Reuse key check for sector: %d",
                 m->key_attack_current_sector);
         } else {
-            snprintf(draw_str, sizeof(draw_str), "Unlocking sector: %d", m->sector_current);
+            snprintf(draw_str, sizeof(draw_str), "Unlocking sector: %d", m->current_sector);
         }
         canvas_draw_str_aligned(canvas, 0, 10, AlignLeft, AlignTop, draw_str);
         float dict_progress = m->dict_keys_total == 0 ?
                                   0 :
                                   (float)(m->dict_keys_current) / (float)(m->dict_keys_total);
         float progress = m->sectors_total == 0 ? 0 :
-                                                 ((float)(m->sector_current) + dict_progress) /
+                                                 ((float)(m->current_sector) + dict_progress) /
                                                      (float)(m->sectors_total);
-        if(progress > 1.0) {
-            progress = 1.0;
+        if(progress > 1.0f) {
+            progress = 1.0f;
         }
         if(m->dict_keys_current == 0) {
             // Cause when people see 0 they think it's broken
-            snprintf(draw_str, sizeof(draw_str), "%d/%d", 1, m->dict_keys_total);
+            snprintf(draw_str, sizeof(draw_str), "%d/%zu", 1, m->dict_keys_total);
         } else {
             snprintf(
-                draw_str, sizeof(draw_str), "%d/%d", m->dict_keys_current, m->dict_keys_total);
+                draw_str, sizeof(draw_str), "%zu/%zu", m->dict_keys_current, m->dict_keys_total);
         }
         elements_progress_bar_with_text(canvas, 0, 20, 128, dict_progress, draw_str);
         canvas_set_font(canvas, FontSecondary);
-        snprintf(draw_str, sizeof(draw_str), "Keys found: %d/%d", m->keys_found, m->keys_total);
+        snprintf(
+            draw_str, sizeof(draw_str), "Keys found: %d/%d", m->keys_found, m->sectors_total * 2);
```

Comment:
self-documented code is better

#define NFC_CLASSIC_KEYS_PER_SECTOR 2
```suggestion
            draw_str, sizeof(draw_str), "Keys found: %d/%d", m->keys_found, m->sectors_total * NFC_CLASSIC_KEYS_PER_SECTOR);
```